### ISSUE TITLE
Can't consume in C# client messages written by java producer.

### ISSUE THREAD

**Issue by mesdocs** [confluentinc/confluent-kafka-dotnet#588]
Description
===========
Based on spring cloud stream examples, I have created a simple java producer that writes kafka messages corresponding to an avro schema stored in the schema registry.
I have also created a simple java consumer that consumes these messages.
All seems to work well. I can produce and consume messages.

Now I try to create a simple C# consumer that consumes the same messages.
I use avrogen to compile avro schema and get C# source code usable in my project.
I pick AvroSpecific in the examples as a base for my consumer. Unfortunately I get an exception on deserialization.

Here is the stack trace :
Consume error: System.IO.InvalidDataException: magic byte should be 0, not 164
   at Confluent.Kafka.Serialization.SpecificDeserializerImpl`1.Deserialize(String topic, Byte[] array)
   at Confluent.Kafka.Serialization.AvroDeserializer`1.Deserialize(String topic,Byte[] data)

Could anyone help to solve this problem ?

Best regards.
Mesdocs.

How to reproduce
================

Checklist
=========
Please provide the following information:

 - [X] Confluent.Kafka nuget version: 0.11.5
 - [X] Apache Kafka version: confluent-4.1.1 Open Source
 - [ ] Client configuration: 
 - [X] Operating system: Windows 7 (
 - [ ] Provide logs (with "debug" : "..." as necessary in configuration)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by mhowlett** [comment]
Are you using `io.confluent.kafka.serializers.KafkaAvroSerializer` to produce your key or value in java? both the Java serializer and the .NET deserializer conform to the wire format documented here: https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html#wire-format in which the first byte is 0.

**Comment by mesdocs** [comment]
Thanks for your answer.
You are right. I had a bad yml configuration file (misspelled key and value serializer properties). The used serializer was the Apache one.
I set KafkaAvroSerializer and schema.registry.url in the java producer.
I set KafkaAvroDeserializer and schema.registry.url in the java consumer.
They seem to work as expected.
My c# consumer does not through the magic byte exception but now throws an AvroException

Consume error: Avro.AvroException: Schema mismatch. Reader: {"type":"record","name":"Flight","namespace":"test.schema.avro","fields":[{"name":"
id","type":"long"},{"name":"airline","type":"string"},{"name":"flightnumber","type":"int"},{"name":"suffix","type":["string","null"]},{"name":"flightdate","type
":"string"},{"name":"departure","type":"string"},{"name":"arrival","type":"string"},{"name":"iscancelled","type":"int"},{"name":"isdeleted","type":"int"},{"name
":"aircraft","type":{"type":"record","name":"Aircraft","namespace":"test.schema.avro","fields":[{"name":"actype","type":"string"},{"name":"registration","type":"string"}]}},{"name":"crewcomposition","type":{"type":"array","items":{"type":"record","name":"CrewPosition","namespace":"test.schema.avro","fields":[{"name":"crewcategory","type":"string"},{"name":"requirednumber","type":"int"}]}}}]}, writer: {"type":"bytes"}
   at Avro.Generic.DefaultReader.Read[T](T reuse, Decoder decoder)
   at Avro.Specific.SpecificReader`1.Read(T reuse, Decoder dec)
   at Confluent.Kafka.Serialization.SpecificDeserializerImpl`1.Deserialize(String topic, Byte[] array)
   at Confluent.Kafka.Serialization.AvroDeserializer`1.Deserialize(String topic,Byte[] data)
   at Confluent.Kafka.Serialization.MessageExtensions.Deserialize[TKey,TValue](Message message, IDeserializer`1 keyDeserializer, IDeserializer`1 valueDeserializer)

What could be the cause of this exception ?

Best regards.
Mesdocs.

**Comment by mhowlett** [comment]
you would get that error if you try to consume a message that was serialized with a schema that isn't compatible with the schema you fed into `avrogen` to generate the class you're using with the deserializer. if you believe the schemas used should be compatible, please provide both and we can comment / debug.

**Comment by mesdocs** [comment]
Thanks you for answer.
I finally solved my problem.
I reinitialized my schema-registry and used a dotnet producer to send messages on kafka. 
Then I used a java producer to send messages. Here I got an error, that pointed me the producer was misconfigured : default byte encoding was processed before the message was processed by the KafkaAvroSerializer class.
I had a look at my java producer configuration and found the spring-cloud.stream use-native-encoding was misplaced.
After correction, both producer are now functional.

Mesdocs.
